spec/controllers/people_controller_spec.rb

```diff
@@ -20,139 +20,30 @@
 
 RSpec.describe PeopleController, type: :controller do
 
-  # This should return the minimal set of attributes required to create a valid
-  # Person. As you add validations to Person, be sure to
-  # adjust the attributes here as well.
-  let(:valid_attributes) {
-    skip("Add a hash of attributes valid for your model")
-  }
+  describe "GET show.json" do
+    let!(:person) { create(:person) }
+    subject(:show_person) { get :show, params: { id: person.id, format: :json } }
 
-  let(:invalid_attributes) {
-    skip("Add a hash of attributes invalid for your model")
-  }
+    it "assign @person " do
+      show_person
+      expect(assigns(:person)).to be_a(Person)
 
-  # This should return the minimal set of values that should be in the session
-  # in order to pass any filters (e.g. authentication) defined in
-  # PeopleController. Be sure to keep this updated too.
-  let(:valid_session) { {} }
 
-  describe "GET #index" do
-    it "assigns all people as @people" do
-      person = Person.create! valid_attributes
-      get :index, params: {}, session: valid_session
-      expect(assigns(:people)).to eq([person])
-    end
-  end
-
-  describe "GET #show" do
-    it "assigns the requested person as @person" do
-      person = Person.create! valid_attributes
-      get :show, params: {id: person.to_param}, session: valid_session
-      expect(assigns(:person)).to eq(person)
-    end
-  end
-
-  describe "GET #new" do
-    it "assigns a new person as @person" do
-      get :new, params: {}, session: valid_session
-      expect(assigns(:person)).to be_a_new(Person)
-    end
-  end
-
-  describe "GET #edit" do
-    it "assigns the requested person as @person" do
-      person = Person.create! valid_attributes
-      get :edit, params: {id: person.to_param}, session: valid_session
-      expect(assigns(:person)).to eq(person)
-    end
-  end
+      expect(assigns(:person).id).to eq(person.id)
```

Comment:
There's no need for this id validation since you're doing it with the other fields below.